src/modules/RA/Scene/components/GeolayerSelect.js

```diff
@@ -91,11 +136,24 @@ const GeolayerSelect = ({ dataProvider, onChange, excludeIds = [], includeIds =
   return (
     <FormControl fullWidth>
       <InputLabel htmlFor="geolayer">Couche</InputLabel>
+
+      {/* Do we have too many element to have perf issues ? */}
+      { (geolayers.length > MAX_NON_NATIVE_SELECT) && (
+      <NativeSelect onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
+        {geolayers.map(({ id, name }) => (
+          <option key={id} value={id}>{name}({id})</option>
+        ))}
+      </NativeSelect>
+      )}
+
+      { (geolayers.length <= MAX_NON_NATIVE_SELECT) && (
       <Select onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
         {geolayers.map(({ id, name }) => (
-          <MenuItem key={id} value={id}>{name}</MenuItem>
+          <MenuItem key={id} value={id}>{name}({id})</MenuItem>
         ))}
       </Select>
+      )}
```

Comment:
As both *selects* receive same props, you may avoid duplicating JSX blocks by using a variable for storing the right component to use:

```jsx
  const SelectComponent = geolayers.length > MAX_NON_NATIVE_SELECT
    ? { Main: NativeSelect, Item: 'option' }
    : { Main: Select, Item: MenuItem };

  /*  OR:

  const SelectComponent = {};
  if (geolayers.length > MAX_NON_NATIVE_SELECT) {
    LocalSelect.Main = NativeSelect;
    LocalSelect.Item = 'option';
  } else {
    LocalSelect.Main = Select;
    LocalSelect.Item = MenuItem;
  }

  */

  return (
    <FormControl fullWidth>
      <InputLabel htmlFor="geolayer">Couche</InputLabel>

      <SelectComponent.Main onChange={handleChoice} {...sanitizeRestProps(props)} name="geolayer">
        {geolayers.map(({ id, name }) => (
          <SelectComponent.Item key={id} value={id}>{name}({id})</SelectComponent.Item>
        ))}
      </SelectComponent.Main>
    </FormControl>
  );
```

Reply:
May be later ;-) but thanks